lib/use_cases/login_use_case.dart

```diff
@@ -15,17 +16,25 @@ class LoginCredential {
   LoginCredential({this.email, this.password});
 }
 
-class LoginUseCase extends UseCase<AuthToken, LoginCredential> {
+class LoginUseCase extends UseCase<void, LoginCredential> {
   final OAuthRepository _repository;
+  final SharedPreferencesStorage _sharedPreferencesStorage;
 
-  const LoginUseCase(this._repository);
+  const LoginUseCase(this._repository, this._sharedPreferencesStorage);
 
   @override
-  Future<Result<AuthToken>> call(LoginCredential credential) {
+  Future<Result<void>> call(LoginCredential credential) {
     return _repository
         .login((credential.email), credential.password)
-        .then<Result<AuthToken>>((value) => Success(value))
+        .then((value) => _persistTokenData(value))
         .onError((err, stackTrace) => Failed(
             UseCaseException(NetworkExceptions.fromDioException(err), err)));
   }
+
+  Result<dynamic> _persistTokenData(AuthToken data) {
+    _sharedPreferencesStorage.saveTokenType(data.tokenType);
+    _sharedPreferencesStorage.saveAccessToken(data.accessToken);
+    _sharedPreferencesStorage.saveRefreshToken(data.refreshToken);
```

Comment:
should we `await` here before return value? 🙏

Reply:
it's not an async function, but inside of each call, when calling to shared_pref `set` method it returns a Future. In this case, I think I should return await inside then? 🤔